internal/tiltfile/tiltfile_test.go

```diff
@@ -4226,6 +4226,48 @@ local_resource('e', 'echo e')
 	}
 }
 
+func TestSetResources(t *testing.T) {
```

Comment:
Can you make this test part of the subpackage Trying to get to a place where the extensions are independent and it's easier to build tooling on them independent of tilt

Reply:
Since you mentioned the test but not the logic, I want to make sure you're aware that this is specifically testing the interaction between the `flags` state and the `tilt up` args. We could move the test into the subpackage, but then that interaction would also need to be moved in, and I feel like that makes more sense in tiltfile proper?

(FWIW, the tests of the independent `flags` behavior are in the `flags` package)

Reply:
Wow, I somehow missed the first sentence of the previous comment.

When working on the next PR after this one, it became apparent that this logic definitely belongs in the extension.